lua/entities/gmod_tardis_interior/modules/cl_vgui.lua

```diff
@@ -1,16 +1,31 @@
 -- VGUI overrides
 
+local function number_ok(pnl, text)
+    return (not pnl:GetNumeric()) or (text == "") or (tonumber(text) ~= nil)
+end
+
 local textpnl
 local function RequestInput( pnl )
     if not textpnl then
         textpnl=pnl
-        Derma_StringRequest(TARDIS:GetPhrase("Common.Interface"),pnl.sub3D2D or pnl.strTooltipText or TARDIS:GetPhrase("Common.EnterTextInput"),pnl:GetText(),function(text)
-            pnl:SetText( text )
-            pnl:SetCaretPos( text:len() )
-            pnl:OnTextChanged()
-            pnl:OnEnter()
-            textpnl=nil
-        end,function() textpnl=nil end)
+        local old_text = pnl:GetText()
+        Derma_StringRequest(TARDIS:GetPhrase("Common.Interface"),
+            pnl.sub3D2D or pnl.strTooltipText or TARDIS:GetPhrase("Common.EnterTextInput"),
+            pnl:GetText(),
+            function(text)
+                if text ~= old_text and number_ok(pnl, text) then
```

Comment:
This element is based on `DTextEntry`
DTextEntry has a numeric option defined `SetNumeric()` / `GetNumeric()` functions
This makes the 3D version work with that

As you can see above, if `pnl` is not numeric, it allows any text:
```lua
local function number_ok(pnl, text)
    return (not pnl:GetNumeric()) or (text == "") or (tonumber(text) ~= nil)
end
```

If the panel has `SetNumeric()` set to true, it will filter out non-numeric values, exactly like it does now.

In the coordinates screen, you can change the name to any text you like, but the coordinates are forced to be numbers, which helps us avoid many errors: we do `x:SetNumeric(true)` for x,y,z,pitch,yaw and roll, but `namebox` doesn't have that

Reply:
Ah okay, makes sense I thought pnl:GetNumeric() was getting the value of the text as a number which is what caused my confusion :)